website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+                if(!target.hasClass('collectionMenu') && !target.hasClass('fa-ellipsis-v') && target.parents('.collection').length === 0) {
+                    self.showCollectionMenu(false);
+                    m.redraw(); // we have to force redraw here
+                }
+            });
+        };
+
+        self.addCollection = function _addCollection () {
+            var url = $osf.apiV2Url('collections/', {});
+            var data = {
+                'data': {
+                    'type': 'collections',
+                    'attributes': {
+                        'title': self.newCollectionName(),
+                    }
+                }
+            };
+            var promise = m.request({method : 'POST', url : url, config : xhrconfig, data : data});
+            promise.then(function(result){
+                var node = result.data;
+                var count = node.relationships.linked_nodes.links.related.meta.count || 0;
+                self.collections().push(new LinkObject('collection', { path : 'collections/' + node.id + '/linked_nodes/', query : { 'related_counts' : true }, systemCollection : false, node : node, count : m.prop(count) }, node.attributes.title));
+                args.sidebarInit();
+            });
+            self.newCollectionName('');
+            self.dismissModal();
+            self.calculateTotalPages();
+            return promise;
+        };
+        self.deleteCollection = function _deleteCollection(){
+            var url = self.collectionMenuObject().item.data.node.links.self;
+            var promise = m.request({method : 'DELETE', url : url, config : xhrconfig});
+            promise.then(function(result){
+                for ( var i = 0; i < self.collections().length; i++) {
+                    var item = self.collections()[i];
+                    if (item.data.node && item.data.node.id === self.collectionMenuObject().item.data.node.id){
+                        self.collections().splice(i, 1);
+                        break;
+                    }
+                }
+            });
+            self.dismissModal();
+            self.calculateTotalPages();
+            return promise;
+        };
+        self.renameCollection = function _renameCollection() {
+            var url = self.collectionMenuObject().item.data.node.links.self;
+            var nodeId = self.collectionMenuObject().item.data.node.id;
+            var title = self.collectionMenuObject().item.renamedLabel;
+            var data = {
+                'data': {
+                    'type': 'collections',
+                    'id':  nodeId,
+                    'attributes': {
+                        'title': title
+                    }
+                }
+            };
+            var promise = m.request({method : 'PATCH', url : url, config : xhrconfig, data : data});
+            promise.then(function(result){
+                self.collectionMenuObject().item.label = title;
+                m.redraw(true);
```

Comment:
If you are trying to get the title to update, I think this will do the trick:

```js
var updatedCollMenuObj = $.extend(true, {}, self.collectionMenuObject(), {item: {label: title}});
self.collectionMenuObject(updatedCollMenuObj);
```

The invocation of `collectionMenuObject` should result in a redraw.